pkg/rbac/sharedData.go

```diff
@@ -32,116 +40,173 @@ type resource struct {
+
+		return true
+	}
+	return false
+}
+
+// Obtain all the cluster-scoped resources in the hub cluster that support list and watch.
+// Equivalent to: `oc api-resources -o wide | grep false | grep watch | grep list`
+func (shared *SharedData) getClusterScopedResources(cache *Cache, ctx context.Context) error {
 
 	// lock to prevent checking more than one at a time and check if cluster scoped resources already in cache
 	shared.csLock.Lock()
 	defer shared.csLock.Unlock()
-	if shared.csResources == nil ||
-		time.Now().After(shared.csUpdatedAt.Add(time.Duration(config.Cfg.SharedCacheTTL)*time.Millisecond)) {
-
-		// if not in cache query database
-		klog.V(6).Info("Querying database for cluster-scoped resources.")
-		shared.csErr = nil // Clear previous errors.
-
-		// Building query to get cluster scoped resources
-		// Original query: "SELECT DISTINCT(data->>apigroup, data->>kind) FROM search.resources WHERE
-		// cluster='local-cluster' AND namespace=NULL"
-
-		schemaTable := goqu.S("search").Table("resources")
-		ds := goqu.From(schemaTable)
-		query, _, err := ds.SelectDistinct(goqu.COALESCE(goqu.L(`"data"->>'apigroup'`), "").As("apigroup"),
-			goqu.COALESCE(goqu.L(`"data"->>'kind_plural'`), "").As("kind")).
-			Where(goqu.L(`"cluster"::TEXT = 'local-cluster'`), goqu.L(`"data"->>'namespace'`).IsNull()).ToSQL()
-		if err != nil {
-			klog.Errorf("Error creating query [%s]. Error: [%+v]", query, err)
-			shared.csErr = err
-			shared.csResources = []resource{}
-			return shared.csResources, shared.csErr
-		}
+	//clear previous cache
+	shared.csResources = nil
+	shared.csErr = nil
+	klog.V(6).Info("Querying database for cluster-scoped resources.")
+
+	// Building query to get cluster scoped resources
+	// Original query: "SELECT DISTINCT(data->>apigroup, data->>kind) FROM search.resources WHERE
+	// cluster='local-cluster' AND namespace=NULL"
+	schemaTable := goqu.S("search").Table("resources")
+	ds := goqu.From(schemaTable)
+	query, _, err := ds.SelectDistinct(goqu.COALESCE(goqu.L(`"data"->>'apigroup'`), "").As("apigroup"),
+		goqu.COALESCE(goqu.L(`"data"->>'kind_plural'`), "").As("kind")).
+		Where(goqu.L(`"cluster"::TEXT = 'local-cluster'`), goqu.L(`"data"->>'namespace'`).IsNull()).ToSQL()
+	if err != nil {
+		klog.Errorf("Error creating query [%s]. Error: [%+v]", query, err)
+		shared.csErr = err
+		shared.csResources = []resource{}
+		return shared.csErr
+	}
+
+	rows, queryerr := cache.pool.Query(ctx, query)
+	if queryerr != nil {
+		klog.Errorf("Error resolving query [%s]. Error: [%+v]", query, queryerr.Error())
+		shared.csErr = queryerr
+		shared.csResources = []resource{}
+		return shared.csErr
+	}
+
+	if rows != nil {
+		defer rows.Close()
+
+		for rows.Next() {
+			var kind string
+			var apigroup string
+			err := rows.Scan(&apigroup, &kind)
+			if err != nil {
+				klog.Errorf("Error %s retrieving rows for query:%s for apigroup %s and kind %s", err.Error(), query,
```

Comment:
Lower to warning because this is a partial error.
```suggestion
				klog.Warnf("Error %s retrieving rows for query:%s for apigroup %s and kind %s", err.Error(), query,
```